### PR TITLE
Add find by github endpoint

### PR DESCRIPTION
**Pull request by kheppenstall**
@neight-allen Although oauth with census is not going to pan out with our react we are hoping to still be able to hit census with a user's github username and get their census data (since we need cohorts).

So our idea was to save their github username and then hit census to retrieve that user's cohort data (and then have a rake task to automatically update cohorts once per module).

I added and tested the following endpoint to census to help us facilitate that so let me know what you think. My main remaining question is how our application could get a census api token to have access to hit the census API.

-Endpoint: `/api/v1/users/find_by_github?q=username`
-Return first match of github username since no two census users should
have the same github username
-Return 404 when no user is found or search params are invalid

### CODE REVIEW

**Comment by kheppenstall** [comment]
@neight-allen Seems like Travis is not going to pass since there are some failing tests on my machine that look like this. Think this is my development setup or just failing tests on the staging branch?

screen shot 2017-04-05 at 10 52 43 am

**Comment by kheppenstall** [comment]
Here is a screenshot of the `find_by_github` tests passing:
screen shot 2017-04-05 at 10 52 59 am

File: config/routes.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@
       namespace :users do
         get '/by_name', to: 'by_name#index'
         get '/by_cohort', to: 'by_cohort#index'
+        get '/find_by_github', to: 'find_by_github#show'
```
Comment (wlffann) [inline config/routes.rb]: Should this be changed to '/by_github' to match the other controllers?

Comment (kheppenstall) [inline config/routes.rb]: @wlffann Maybe yea. Those are ally `find_all` and this one is just `find` so I wasn't sure how to try to differentiate that.

Comment (kheppenstall) [inline config/routes.rb]: Thoughts on how we could try to differentiate that with the route?

Comment (kheppenstall) [inline config/routes.rb]: @wlffann Made that change to be `by_github`

File: app/controllers/api/v1/users/by_github_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Api::V1::Users::ByGithubController < Api::V1::ApiController
+  before_action :doorkeeper_authorize!
+
+  def show
+    if params[:q] && user = User.find_by(git_hub: params[:q])
+      render json: user, serializer: UserSerializer, status: 200
+    else
+      render json: "User not found with github username #{params[:q]}", status: 404
+    end
```
Comment (neight-allen) [inline app/controllers/api/v1/users/by_github_controller.rb]: JSON should always be an object or an array. Just rendering a string isn't really JSON. In fact, it's not clear to me what is actually in the body of this response.

Modify this to be something like 
```
render json: { error: "User not found...." }
```

Comment (kheppenstall) [inline app/controllers/api/v1/users/by_github_controller.rb]: @neight-allen That makes sense. Changed it to 
```ruby
render json: { error: "User not found" }, status: 404
```

**Comment by neight-allen** [comment]
Since this is merged, it should deploy to staging soon. I'll get it to production later.

**Pull request merged**